src/Naneau/SemVer/Version.php

```diff
@@ -132,6 +133,93 @@ public function hasBuild()
     }
 
     /**
+     * Get the next logical version relative to the provided base version. If
+     * no base is supplied, base will be the same as the current version.
+     *
+     * @param  Version|string|null $base
+     * @return Version
+     */
+    public function next(/* $base */)
```

Comment:
I think I would prefer it if `next()` were to always return a new SemVer instance, this could probably be done with a simple clone in `ensureBase()`